### DISCUSSION TITLE
Help! Terraform Cloud have removed `python` from their worker image

### DISCUSSION THREAD

**Discussion by KOConchobhair** [gruntwork-io/knowledge-base#412] [Help] [answered]
I have been using Gruntwork modules in conjunction with Terraform Cloud for over a year. Last night, when I went to change my AMI ID to trigger a rolling deployment in the ASG module, I got the following errors:
![image](https://user-images.githubusercontent.com/2014215/167910290-8bcb3c10-53ad-4c14-a062-aa6ca6ae410b.png)

So it appears they have removed `python` executable from their TF cloud worker/runner. As far as I can tell, there was no notice, no documentation or release notes on this - they just changed it without warning and now my deployment is completely blocked. I contacted HashiCorp support and they simply said
> You are correct that there were some changes made to the workers' image.
> Currently, python3 is available on the image.
> Would it work for you to substitute python with python3?

So I ask the grunts, is it possible to use `python3` instead of `python` for this script?
https://github.com/gruntwork-io/terraform-aws-asg/blob/457efad3ea8a0682c8bc01d0696f4b75cfeb3314/modules/server-group/main.tf#L635

And update the version of join-path module used to get support for `python3`?
https://github.com/gruntwork-io/terraform-aws-ecs/blob/4d156d23ccfd672532d3c7ed323fbc4cdf97df5f/modules/ecs-service/deployment_check.tf#L78
https://github.com/gruntwork-io/terraform-aws-asg/blob/457efad3ea8a0682c8bc01d0696f4b75cfeb3314/modules/server-group/main.tf#L668

I know Gruntwork may not provide explicit support for running on Terraform Cloud but I would think it benefits everyone if the awesome Gruntwork modules work in that environment as well. 

P.S. There's an additional error (same cause) due to python usage in the ecs_deployment_check module as well:
![image](https://user-images.githubusercontent.com/2014215/167910418-253b386b-6ada-47ae-951c-3d499460f355.png)
In this case, I even have `enable_ecs_deployment_check = false` but it still runs the python code.

---

<ins datetime="2022-05-11T17:49:20Z">
  <p><a href="https://support.gruntwork.io/hc/requests/108578">Tracked in ticket #108578</a></p>
</ins>

**Comment by yorinasub17** [comment, marked as answer]
Hi we are in the process of implementing this. We recently updated `terraform-aws-utilities` to use `python3` instead of `python`, and are working towards flowing that update through the system. We can't make commitments on when that would be done, but please be on the look out for releases on the relevant repos where we will mention that this change has been done.
